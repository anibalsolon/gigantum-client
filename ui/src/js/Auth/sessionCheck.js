// vendor
import { graphql } from 'react-relay';
// mutations
import RemoveUserIdentityMutation from 'Mutations/RemoveUserIdentityMutation';
// environment
import { fetchQuery } from 'JS/createRelayEnvironment';

const sessionCheckQuery = graphql`
  query sessionCheckQuery{
    userIdentity{
      isSessionValid
    }
  }
`;

const UserIdentity = {
  /*
    fetches session information from useridentity field in api
  */
  getUserIdentity: () => new Promise((resolve, reject) => {
    const fetchData = function () {
      fetchQuery(sessionCheckQuery(), {}).then((response, error) => {
        if (response) {
          resolve(response);
        } else {
          reject(response);
        }
      }).catch((error) => {
        reject(error);
      });
    };

    fetchData();
  }),
};

export default UserIdentity;
