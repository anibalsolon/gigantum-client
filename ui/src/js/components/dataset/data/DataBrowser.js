// vendor
import React, { Component } from 'react';
import { createPaginationContainer, graphql } from 'react-relay';
// mutations
import FileBrowserWrapper from 'Components/labbook/fileBrowser/FileBrowserWrapper';
// store
import store from 'JS/redux/store';

class DataBrowser extends Component {
  constructor(props) {
  	super(props);

    this.state = {
      rootFolder: '',
      moreLoading: false,
    };

    this.setRootFolder = this.setRootFolder.bind(this);
    this._loadMore = this._loadMore.bind(this);
  }

  /*
    loads more if branches are switched
  */
  componentDidUpdate() {
    this.props.loadStatus(this.state.moreLoading);
    if (!this.state.moreLoading && this.props.data.allFiles && this.props.data.allFiles.edges.length < 3 && this.props.data.allFiles.pageInfo.hasNextPage) {
      this._loadMore();
    }
  }

  /*
    handle state and addd listeners when component mounts
  */
  componentDidMount() {
    this.props.loadStatus(this.state.moreLoading);
    if (this.props.data.allFiles &&
      this.props.data.allFiles.pageInfo.hasNextPage) {
      this._loadMore(); // routes query only loads 2, call loadMore
    } else {
      this.setState({ moreLoading: false });
    }
  }
  /*
    @param
    triggers relay pagination function loadMore
    increments by 100
    logs callback
  */
  _loadMore() {
    this.setState({ moreLoading: true });
    const self = this;
    this.props.relay.loadMore(
      100, // Fetch the next 100 feed items
      (response, error) => {
        if (error) {
          console.error(error);
        }

        if (self.props.data.allFiles &&
         self.props.data.allFiles.pageInfo.hasNextPage) {
          self._loadMore();
        } else {
          this.setState({ moreLoading: false });
        }
      },
    );
  }
  /*
    @param
    sets root folder by key
    loads more files
  */
  setRootFolder(key) {
    this.setState({ rootFolder: key });
  }

  render() {
    if (this.props.data && this.props.data.allFiles) {
      let dataFiles = this.props.data.allFiles;
      if (this.props.data.allFiles.edges.length === 0) {
        dataFiles = {
          edges: [],
          pageInfo: this.props.data.allFiles.pageInfo,
        };
      }

      return (
        <FileBrowserWrapper
          ref="dataBrowser"
          section="data"
          selectedFiles={this.props.selectedFiles}
          clearSelectedFiles={this.props.clearSelectedFiles}
          setRootFolder={this.setRootFolder}
          files={dataFiles}
          parentId={this.props.dataId}
          connection="DataBrowser_allFiles"
          favoriteConnection="DataFavorites_favorites"
          favorites={this.props.favorites}
          isLocked={{ locked: false }}
          {...this.props}
        />
      );
    }
    return (<div>No Files Found</div>);
  }
}

export default createPaginationContainer(
  DataBrowser,
  {

    data: graphql`
      fragment DataBrowser_data on Dataset{
        allFiles(after: $cursor, first: $first)@connection(key: "DataBrowser_allFiles", filters: []){
          edges{
            node{
              id
              isDir
              isFavorite
              modifiedAt
              key
              size
            }
            cursor
          }
          pageInfo{
            hasNextPage
            hasPreviousPage
            startCursor
            endCursor
          }
        }
      }`,
  },
  {
    direction: 'forward',
    getConnectionFromProps(props) {
      return props.data && props.data.allFiles;
    },
    getFragmentVariables(prevVars, totalCount) {
      return {
        ...prevVars,
        first: totalCount,
      };
    },
    getVariables(props, { count, cursor }, fragmentVariables) {
      const { owner, labbookName } = store.getState().routes;

      return {
        first: count,
        cursor,
        owner,
        name: labbookName,
      };
    },
    query: graphql`
      query DataBrowserPaginationQuery(
        $first: Int
        $cursor: String
        $owner: String!
        $name: String!
      ) {
        dataset(name: $name, owner: $owner){
           id
           description
           ...DataBrowser_data
        }
      }
    `,
  },

);
