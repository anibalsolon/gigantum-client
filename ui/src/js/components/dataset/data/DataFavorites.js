// // vendor
// import React, { Component } from 'react';
// import { createPaginationContainer, graphql } from 'react-relay';
// // componenets
// import DataFavoriteList from './DataFavoriteList';
// import FileEmpty from 'Components/dataset/overview/FileEmpty';
// // store
// import store from 'JS/redux/store';

// class DataFavorites extends Component {
//   constructor(props) {
//   	super(props);
//     this.state = {
//       loading: false,
//     };
//   }
//   /*
//     update component when props are reloaded
//   */
//   UNSAFE_componentWillReceiveProps(nextProps) {
//     // this._loadMore() //routes query only loads 2, call loadMore
//     if (nextProps.data && nextProps.data.favorites && nextProps.data.favorites.pageInfo.hasNextPage && nextProps.data.favorites.edges.length < 3) {
//       this.props.relay.loadMore(
//         1, // Fetch the next 10 feed items
//         (response, error) => {
//           if (error) {
//             console.error(error);
//           }
//         },
//       );
//     }
//   }

//   /**
//     handle state and addd listeners when component mounts
//   */
//   componentDidMount() {
//     // this._loadMore() //routes query only loads 2, call loadMore
//     if (this.props.data && this.props.data.favorites && this.props.data.favorites.pageInfo.hasNextPage && this.props.data.favorites.edges.length < 3) {
//       this.props.relay.loadMore(
//         1, // Fetch the next 10 feed items
//         (response, error) => {
//           if (error) {
//             console.error(error);
//           }
//         },
//       );
//     }
//   }

//   /**
//     @param
//     triggers relay pagination function loadMore
//     increments by 10
//     logs callback
//   */
//   _loadMore() {
//     const self = this;

//     this.setState({ loading: true });

//     this.props.relay.loadMore(
//       3, // Fetch the next 10 feed items
//       (response, error) => {
//         self.setState({ loading: false });

//         if (error) {
//           console.error(error);
//         }
//       },
//     );
//   }


//   render() {
//     if (this.props.data && this.props.data.favorites) {
//       let loadingClass = (this.props.data.favorites.pageInfo.hasNextPage) ? 'Favorite__action-bar' : 'hidden';
//       loadingClass = (this.state.loading) ? 'Favorite__action-bar--loading' : loadingClass;

//       if (this.props.data.favorites.edges.length > 0) {
//         const favorites = this.props.data.favorites.edges.filter(edge => edge && (edge.node !== undefined));

//         return (

//           <div className="Favorite">
//             <DataFavoriteList
//               datasetName={this.props.datasetName}
//               dataId={this.props.dataId}
//               section="data"
//               favorites={favorites}
//               owner={this.props.owner}
//             />


//             <div className={loadingClass}>
//               <button
//                 className="Favorite__load-more"
//                 onClick={() => { this._loadMore(); }}
//               >
//                 Load More
//               </button>
//             </div>
//           </div>

//         );
//       }
//       return (
//         <FileEmpty
//           section="data"
//           mainText="This Project has No Data Favorites"
//         />
//       );
//     }
//     return (<div>No Files Found</div>);
//   }
// }

// export default createPaginationContainer(
//   DataFavorites,
//   {

//     data: graphql`
//       fragment DataFavorites_data on DatasetFileConnection{
//         favorites(after: $cursor, first: $first)@connection(key: "DataFavorites_favorites"){
//           edges{
//             node{
//               id
//               owner
//               name
//               index
//               key
//               description
//               isDir
//               associatedDatasetFileId
//               section
//             }
//             cursor
//           }
//           pageInfo{
//             hasNextPage
//             hasPreviousPage
//             startCursor
//             endCursor
//           }
//         }
//       }`,
//   },
//   {
//     direction: 'forward',
//     getConnectionFromProps(props) {
//       return props.data && props.data.favorites;
//     },
//     getFragmentVariables(prevVars, totalCount) {
//       return {
//         ...prevVars,
//         first: totalCount,
//       };
//     },
//     getVariables(props, { count, cursor }, fragmentVariables) {
//       const { owner, datasetName } = store.getState().routes;
//       return {
//         first: count,
//         cursor,
//         owner,
//         name: datasetName,
//       };
//     },
//     query: graphql`
//       query DataFavoritesPaginationQuery(
//         $first: Int
//         $cursor: String
//         $owner: String!
//         $name: String!
//       ) {
//         dataset(name: $name, owner: $owner){
//            id
//            description
//            # You could reference the fragment defined previously.
//            ...DataFavorites_data
//         }
//       }
//     `,
//   },

// );
