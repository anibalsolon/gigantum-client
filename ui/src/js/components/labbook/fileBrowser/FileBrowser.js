// vendor
import React, { Component } from 'react';
import store from 'JS/redux/store';
import { DropTarget } from 'react-dnd';
import { NativeTypes } from 'react-dnd-html5-backend';
import classNames from 'classnames';
// assets
import './FileBrowser.scss';
// components
import File from './fileRow/File';
import Folder from './fileRow/Folder';
import AddSubfolder from './fileRow/AddSubfolder';
import FileBrowserMutations from './utilities/FileBrowserMutations';
import Connectors from './utilities/Connectors';

class FileBrowser extends Component {
    constructor(props) {
      super(props);

      this.state = {
        mutations: new FileBrowserMutations(this._getMutationData()),
        mutationData: this._getMutationData(),
        hoverId: '',
        childrenState: {},
        multiSelect: 'none',
        search: '',
        isOverChildFile: false,
      };

      this._deleteSelectedFiles = this._deleteSelectedFiles.bind(this);
      this._setState = this._setState.bind(this);
      this._updateChildState = this._updateChildState.bind(this);
      this._checkChildState = this._checkChildState.bind(this);
      this._updateDropZone = this._updateDropZone.bind(this);
    }
    static getDerivedStateFromProps(props, state) {
        let childrenState = {};
        props.files.edges.forEach((edge) => {
          if (edge.node && edge.node.key) {
            let key = edge.node.key.split('/').join('');
            childrenState[key] = {
              isSelected: (state.childrenState && state.childrenState[key]) ? state.childrenState[key].isSelected : false,
              edge,
            };
          }
        });

        return {
          ...state,
          childrenState,
        };
    }
    /**
    *  @param {string} key - key of file to be updated
    *  @param {boolean} isSelected - update if the value is selected
    *  @return {}
    */
    _updateChildState(key, isSelected) {
      let isChildSelected = false;
      let count = 0;
      let selectedCount = 0;
      let { childrenState } = this.state;
      let santizedKey = key.split('/').join('');
      childrenState[santizedKey].isSelected = isSelected;

      for (let key in childrenState) {
        if (childrenState[key]) {
          if (childrenState[key].isSelected) {
            isChildSelected = true;
            selectedCount++;
          }
          count++;
        }
      }

      let multiSelect = !isChildSelected ? 'none' : (selectedCount === count) ? 'all' : 'partial';

      this.setState({ childrenState, multiSelect });
    }
    /**
    *  @param {}
    *  update state of component for a given key value pair
    *  @return {}
    */
    _setState(stateKey, value) {
       this.setState({ [stateKey]: value });
    }
    /**
    *  @param {}
    *  sorts files into an object for rendering
    *  @return {}
    */
    _processFiles() {
        let edges = this.props.files.edges;
        let edgesToSort = JSON.parse(JSON.stringify(edges));
        let fileObject = {};
        const { search } = this.state;
        const searchLowerCase = search.toLowerCase();

        if (search !== '') {
          let edgesSearchMatch = edgesToSort.filter((edge) => {
            const lowerCaseKey = edge.node.key.toLowerCase();
            return (lowerCaseKey.indexOf(searchLowerCase) > -1);
          });

          edgesToSort = edgesToSort.filter((edge) => {
            let keyMatch = false;
            edgesSearchMatch.forEach((matchEdge) => {
              if (matchEdge.node.key.indexOf(edge.node.key) > -1) {
                keyMatch = true;
              }
            });
            return keyMatch;
          });
        }

        edgesToSort.forEach((edge, index) => {
            let key = edge.node.key.toLowerCase();
            let searchLowerCase = search.toLowerCase();


            if (edge.node) {
              let currentObject = fileObject;
              let splitKey = edge.node.key.split('/').filter(key => key.length);

              splitKey.forEach((key, index) => {
                  if (currentObject && (index === (splitKey.length - 1))) {
                      if (!currentObject[key]) {
                        currentObject[key] = {
                          edge,
                          index,
                        };
                      } else {
                        currentObject[key].edge = edge;
                      }
                  } else if (currentObject && !currentObject[key]) {
                      currentObject[key] = {
                        children: {},
                      };
                      currentObject = currentObject[key].children;
                  } else if (currentObject && currentObject[key] && !currentObject[key].children) {
                      currentObject[key].children = {};
                      currentObject = currentObject[key].children;
                  } else {
                    currentObject = currentObject[key].children;
                  }
              });
           }
        });
        return fileObject;
  }
  /**
  *  @param {}
  *  sorts files into an object for rendering
  *  @return {object}
  */
  _getMutationData() {
    const {
      parentId,
      connection,
      favoriteConnection,
      section,
    } = this.props;
    const { owner, labbookName } = store.getState().routes;

    return {
      owner,
      labbookName,
      parentId,
      connection,
      favoriteConnection,
      section,
    };
  }
  /**
  *  @param {}
  *  loops through selcted files and deletes them
  *  @return {}
  */
  _deleteSelectedFiles() {
    let self = this;
    let filePaths = [];
    let dirList = [];
    let comparePaths = [];
    let edges = [];

    for (let key in this.state.childrenState) {
      if (this.state.childrenState[key].isSelected) {
        let { edge } = this.state.childrenState[key];
        delete this.state.childrenState[key];
        comparePaths.push(edge.node.key);
        filePaths.push(edge.node.key);
        edges.push(edge);
        if (edge.node.isDir) {
          dirList.push(edge.node.key);
        }
      }
    }

    let filteredPaths = filePaths.filter((key) => {
      let folderKey = key.substr(0, key.lastIndexOf('/'));
      folderKey = `${folderKey}/`;

      let hasDir = dirList.some(dir => ((key.indexOf(dir) > -1) && (dir !== key)));
      return !hasDir;
    });
    self._deleteMutation(filteredPaths, edges);
  }

  /**
  *  @param {}
  *  selects all or unselects files
  *  @return {}
  */
  _selectFiles() {
    let isSelected = false;
    let count = 0;
    let selectedCount = 0;

    for (let key in this.state.childrenState) {
      if (this.state.childrenState[key]) {
        if (this.state.childrenState[key].isSelected) {
          isSelected = true;
          selectedCount++;
        }
        count++;
      }
    }
    let multiSelect = (count === selectedCount) ? 'none' : 'all';

    let { childrenState } = this.state;

    for (let key in childrenState) {
      if (childrenState[key]) {
        childrenState[key].isSelected = (multiSelect === 'all');
        count++;
      }
    }
    this.setState({ multiSelect, childrenState });
  }

  /**
  *  @param {}
  *  triggers delete muatation
  *  @return {}
  */
  _deleteMutation(filePaths, edges) {
    const data = {
      filePaths,
      edges,
    };
    this.setState({ multiSelect: 'none' });
    this.state.mutations.deleteLabbookFiles(data, (response) => {});
  }
  /**
  *  @param {string, boolean}
  *  updates boolean state of a given key
  *  @return {}
  */
  _updateStateBoolean(key, value) {
    this.setState({ [key]: value });
  }
  /**
  *  @param {}
  *  checks if folder refs has props.isOver === true
  *  @return {boolean}
  */
  _checkRefs() {
    let isOver = this.props.isOverCurrent || this.props.isOver, // this.state.isOverChildFile,
        { refs } = this;

        Object.keys(refs).forEach((childname) => {
          if (refs[childname].getDecoratedComponentInstance && refs[childname].getDecoratedComponentInstance() && refs[childname].getDecoratedComponentInstance().getDecoratedComponentInstance && refs[childname].getDecoratedComponentInstance().getDecoratedComponentInstance()) {
            const child = refs[childname].getDecoratedComponentInstance().getDecoratedComponentInstance();
            if (child.props.data && !child.props.data.edge.node.isDir) {
              if (child.props.isOverCurrent) {
                isOver = true;
              }
            }
          }
        });
    return ({
      isOver,
    });
  }
  /**
  *  @param {}
  *  checks if folder refs has props.isOver === true
  *  @return {boolean} isSelected - returns true if a child has been selected
  */
  _checkChildState() {
    let isSelected = false;

    for (let key in this.state.childrenState) {
      if (this.state.childrenState[key].isSelected) {
        isSelected = true;
      }
    }

    return { isSelected };
  }
  /**
  *  @param {evt}
  *  update state
  *  @return {}
  */
  _updateSearchState(evt) {
    this.setState({ search: evt.target.value });
  }

  /**
  *  @param {boolean} isOverChildFile
  *  update state to update drop zone
  *  @return {}
  */
  _updateDropZone(isOverChildFile) {
    this.setState({ isOverChildFile });
  }

  render() {
    const files = this._processFiles(),
          { mutationData } = this.state,
          { isOver } = this.props;

   const { isSelected } = this._checkChildState();

   const fileBrowserCSS = classNames({
           FileBrowser: true,
           'FileBrowser--highlight': isOver,
         }),
         deleteButtonCSS = classNames({
           'Btn Btn--round Btn--delete': true,
           hidden: !isSelected,
         }),
         multiSelectButtonCSS = classNames({
           'Btn Btn--round': true,
           'Btn--check': this.state.multiSelect === 'all',
           'Btn--uncheck': this.state.multiSelect === 'none',
           'Btn--partial': this.state.multiSelect === 'partial',
         });

   return (
       this.props.connectDropTarget(<div className={fileBrowserCSS}>

                <div className="FileBrowser__tools flex justify--space-between">
                  <div className="FileBrowser__multiselect flex justify--start">
                    <button
                      className={multiSelectButtonCSS}
                      onClick={() => { this._selectFiles(); }} />
                    <button
                      className={deleteButtonCSS}
                      onClick={() => { this._deleteSelectedFiles(); }} />
                  </div>
                  <div className="FileBrowser__search flex-1">
                    <input
                      className="FileBrowser__input full--border"
                      type="text"
                      placeholder="Search Files Here"
                      onChange={(evt) => { this._updateSearchState(evt); } }
                      onKeyUp={(evt) => { this._updateSearchState(evt); } }
                    />
                  </div>
                </div>
                <div className="FileBrowser__header">
                    <div className="FileBrowser__header--name flex justify--start">
                      <div className="FileBrowser__name-text">
                        File
                      </div>
                    </div>

                    <div className="FileBrowser__header--size">
                        Size
                    </div>

                    <div className="FileBrowser__header--date">
                        Modified
                    </div>

                    <div className="FileBrowser__header--menu">
                    </div>
                </div>
            <div className="FileBrowser__body">
                <AddSubfolder
                  key={'rootAddSubfolder'}
                  folderKey=""
                  mutationData={mutationData}
                  mutations={this.state.mutations}
                />
                {
                    Object.keys(files).map((file) => {
                        if (files[file] && files[file].edge && files[file].edge.node.isDir) {
                            return (
                                <Folder
                                    ref={file}
                                    filename={file}
                                    key={files[file].edge.node.key}
                                    multiSelect={this.state.multiSelect}
                                    mutationData={mutationData}
                                    data={files[file]}
                                    mutations={this.state.mutations}
                                    setState={this._setState}
                                    rowStyle={{}}
                                    updateChildState={this._updateChildState}>
                                </Folder>
                            );
                        } else if (files[file] && files[file].edge && !files[file].edge.node.isDir) {
                          return (
                              <File
                                  ref={file}
                                  filename={file}
                                  key={files[file].edge.node.key}
                                  multiSelect={this.state.multiSelect}
                                  mutationData={mutationData}
                                  data={files[file]}
                                  mutations={this.state.mutations}
                                  expanded
                                  isOverChildFile={this.state.isOverChildFile}
                                  updateParentDropZone={this._updateDropZone}
                                  updateChildState={this._updateChildState}>

                              </File>
                          );
                        }

                        return (<div>Loading</div>);
                    })
                }
            </div>
            {/* <div className={drogTargetCSS}>
              <h3 className="FileBrowser__h3">Add file to root</h3>
            </div> */}
        </div>)
    );
  }
}


export default DropTarget(
    ['card', NativeTypes.FILE],
    Connectors.targetSource,
    Connectors.targetCollect,
  )(FileBrowser);
