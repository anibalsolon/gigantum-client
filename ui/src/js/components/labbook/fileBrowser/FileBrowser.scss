@import '~Styles/_imports.scss';

.FileBrowser {
    width: 100%;
    padding: 20px;
    position: relative;
    z-index: 1;

    &--highlight{
      background-color: rgba(0,126,168, 0.5);
      border: 1px solid $azure;

      .AddSubfolder {
        background-color: transparent !important;
      }
    }
}

.FileBrowser__tools {
  align-items: center;
}

.FileBrowser__multiselect {
  background: #f0f1f5;
  border-radius: 12px;
}

.FileBrowser__search {
  max-width: 500px;
}

.FileBrowser__input{
  margin: 0px !important;
  height: 34px;
}

.FileBrowser__name-text {

}

.FileBrowser__header {
    @include flex(space-between, row);
    align-items: center;

    border-bottom: 2px solid $romanSilver;
    height: 50px;
    width: 100%;
    font-weight: bold;
    font-size: 18px;
    text-align: left;


    &--name{
        margin: 0 60px;
        flex: 1;

    }

    &--size{
        width: 100px;
    }

    &--date{
        width: 140px;
    }

    &--menu{
        width: 140px;
    }
}

.FileBrowser__body {
  overflow-x: visible;
}

.FileBrowser__btn {
  margin: 0 43px 0 8px;
  background: $azure url(~Images/icons/trash-white.svg) no-repeat center center;
  background-size: 20px;

  &:focus {
    background: $azure url(~Images/icons/trash-white.svg) no-repeat center center;
    background-size: 20px;
  }

  &:hover {
    background: $turquise url(~Images/icons/trash-white.svg) no-repeat center center;
    background-size: 20px;
  }
}

.FileBrowser__dropTarget{
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;

  z-index: 1;

  background-color: $azure;

  opacity: 0.7;

  @include flex(center, column);
}

.FileBrowser__h3{
  color: $white;
  font-weight: 100;
  text-align: center;
}
