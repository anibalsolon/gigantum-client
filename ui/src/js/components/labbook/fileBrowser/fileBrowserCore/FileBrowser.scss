@import '~Styles/_imports.scss';

.FileBrowser{
    width: 100%;
    height: 400px;
    overflow-x: auto;
}
.FileBrowser__header{
    @include flex(space-around, row);
}
