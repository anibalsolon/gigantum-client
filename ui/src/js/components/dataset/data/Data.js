// vendor
import React, { Component } from 'react';
import { createFragmentContainer, graphql } from 'react-relay';
// components
import DataBrowser from './DataBrowser';
// import DataFavorites from './DataFavorites';
// import MostRecent from 'Components/datasets/filesShared/MostRecentData';
import ToolTip from 'Components/shared/ToolTip';

class Data extends Component {
  constructor(props) {
  	super(props);
    this.state = {
      selectedFiles: [],
      // selectedFilter: 'favorites',
    };
    this._setSelectedFiles = this._setSelectedFiles.bind(this);
    this._clearSelectedFiles = this._clearSelectedFiles.bind(this);
    this._loadStatus = this._loadStatus.bind(this);
    this._selectFilter = this._selectFilter.bind(this);
  }
  componentDidUpdate() {
    // this.refs[this.state.selectedFilter].classList.add('Data__filter--selected');
    // for (const key in this.refs) {
    //   if (key !== this.state.selectedFilter) {
    //     this.refs[key].classList.remove('Data__filter--selected');
    //   }
    // }
  }

  _setSelectedFiles(evt) {
    const files = [...evt.target.files];
    this.setState({ selectedFiles: files });
  }

  _clearSelectedFiles() {
    this.setState({ selectedFiles: [] });
  }

  _loadStatus(res) {
    if (res !== this.state.loadingStatus) {
      this.setState({ loadingStatus: res });
    }
  }
  _selectFilter(filterName) {
    this.setState({ selectedFilter: filterName });
  }

  render() {
    if (this.props.datasets) {
      return (

        <div className="Data">
          <div className="Data__header">
            <div className="Data__subtitle-container">
              <h5 className="Data__subtitle">Data Browser
                <ToolTip section="dataBrowser" />
                {
                  this.state.loadingStatus &&
                  <div className="Data__loading" />
                }
              </h5>
              <p className="Data__subtitle-sub">Currently only files under 1.8GB are supported.</p>
            </div>

            <div className="Data__toolbar end">
              <p className="Data__import-text" id="Data__">
                <label
                  className="Data__import-file"
                  htmlFor="file__data"
                >
                  Upload File
                </label>
                <input
                  id="file__data"
                  className="hidden"
                  type="file"
                  onChange={(evt) => { this._setSelectedFiles(evt); }}
                />
                or Drag and Drop File Below
              </p>
            </div>
          </div>

          <div className="Data__file-browser">
            <DataBrowser
              selectedFiles={this.state.selectedFiles}
              clearSelectedFiles={this._clearSelectedFiles}
              datasetsId={this.props.datasetsId}
              dataId={this.props.datasets.id}
              data={this.props.datasets}
              loadStatus={this._loadStatus}
              type={this.props.type}
            />
          </div>
        </div>
      );
    }
    return (<div>No Files Found</div>);
  }
}


export default createFragmentContainer(
  Data,
  graphql`
    fragment Data_datasets on Dataset{
      id
      ...DataBrowser_data
    }
  `,
);
