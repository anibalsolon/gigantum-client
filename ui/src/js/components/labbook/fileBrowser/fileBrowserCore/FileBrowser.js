// vendor
import React, { Component } from 'react';
// assets
import './FileBrowser.scss';
// components
import File from './File';
import Folder from './Folder';

export default class FileBrowser extends Component {

    _processFiles() {
        const files = this.props.files;
        let collectedFiles = {};
        this.props.files.forEach((file) => {
            let splitKey = file.key.split('/', -1).filter(key => key.length);
            let currentFileObjectPosition = collectedFiles;
            while (splitKey.length > 1) {
                const currentKey = splitKey[0]
                if (!currentFileObjectPosition[currentKey]) {

                    currentFileObjectPosition[currentKey] = { children: {} };
                }
                currentFileObjectPosition = currentFileObjectPosition[currentKey].children;
                splitKey = splitKey.slice(1, splitKey.length);
            }
            if (splitKey.length === 1) {
                if (currentFileObjectPosition[splitKey[0]]) {
                    currentFileObjectPosition[splitKey[0]].data = file;
                } else {
                    currentFileObjectPosition[splitKey[0]] = { data: file };
                }
                if (file.isDir) {
                    currentFileObjectPosition[splitKey[0]].children = {};
                }
            }
        });
        return collectedFiles;
    }
    _renderJSX(files) {
        return Object.keys(files).map((file) => {
            if (files[file].children) {
                return (<Folder data={files[file]}></Folder>);
            }
            return (<File data={files[file]}></File>);
        })
    }
  render() {
    const files = this._processFiles()
    console.log(files)
    return (
        <div className="FileBrowser">
            <div className="FileBrowser__header">
                <div>
                    File
                </div>

                <div>
                    Size
                </div>

                <div>
                    Modified
                </div>
            </div>
            <div className="FileBrowser__body">
                {
                    Object.keys(files).map((file) => {
                        if (files[file].children) {
                            return (
                                <Folder
                                    data={files[file]}
                                    key={files[file].data.key}
                                >
                                </Folder>
                            );
                        }
                        return (
                            <File
                                data={files[file]}
                                key={files[file].data.key}
                            >
                            </File>
                        );
                    })
                }
            </div>
        </div>
    );
  }
}
