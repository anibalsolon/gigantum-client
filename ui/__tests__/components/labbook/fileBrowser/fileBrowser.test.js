import React from 'react';
import renderer from 'react-test-renderer';

import FileBrowser from 'Components/labbook/fileBrowser/FileBrowser';
import { mount, shallow } from 'enzyme';
import Auth from 'JS/Auth/Auth';
import history from 'JS/history';
import { BrowserRouter as Router } from 'react-router-dom';
import data from './../filesShared/__relayData__/MostRecentCode.json';
const auth = new Auth();

const variables = { first: 20, labbook: 'demo-lab-book' };
export default variables;

const { labbook } = data.data;
const fixtures = {
  clearSelectedFiles: jest.fn(),
  loadStatus: jest.fn(),
  selectedFiles: [],
  labbookId: labbook.id,
  sectionId: labbook.code.id,
  section: 'code',
  isLocked: false,
  ...labbook.code,
};

auth.isAuthenticated = function () { return true; };

describe('FileBrowser component', () => {
  it('Test FileBrowser Rendering', () => {
        const component = renderer.create(<FileBrowser
              {...fixtures}
            />);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
   });

   it('Test FileBrowser Rendering', () => {
     const localLabbooks = mount(<FileBrowser {...fixtures}/>);

     let tree = component.toJSON();
     expect(tree).toMatchSnapshot();
    });
});
