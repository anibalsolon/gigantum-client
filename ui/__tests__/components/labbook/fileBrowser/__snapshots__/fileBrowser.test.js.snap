// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileBrowser component Test FileBrowser Rendering 1`] = `null`;
